### PR TITLE
adds special respawns for admins

### PR DESCRIPTION
**Pull request by corp-0**
### Purpose
This PR will allow admins to respawn any dead player but now changing their occupation and gear to any occupation declared in a list of jobs for admins.

### Notes:
- Adds Centcomm Commander, Centcomm Officer, Centcomm Intern and Death Squads occupations. Only way to select them is by this new button.
- Only dead players can be respawned and their body will keep existing in the world (just like regular respawn).
- Once the job is changed, you can't go back to your normal job.
- Players will respawn at their designed spawnpoints (arrivals for the jobs that have no spawn point). I have no idea how this will work with subscenes, but eventually Death Squads and Centcomm should spawn at Centcomm.
- Tested in editor. I can't test in MP because builds are broken (Hologram shader).

Some pictures:
image
image
image
image
image
Adding new jobs is just as easy as adding Occupations to this list.

### CODE REVIEW

File: UnityProject/Assets/Scripts/Player/PlayerNetworkActions.cs [hunk 1]
```diff
@@ -474,8 +474,13 @@ public void CmdRespawnPlayer()
 	}
 
 	[Server]
-	public void ServerRespawnPlayer()
+	public void ServerRespawnPlayer(string occupation = null)
 	{
+		if (occupation != null)
+		{
+			playerScript.mind.occupation = Resources.Load<Occupation>($"ScriptableObjects/Occupations/"+occupation);
```
Comment (Bod9001) [inline UnityProject/Assets/Scripts/Player/PlayerNetworkActions.cs]: could you use the SingletonScriptableObject to get the references from, you will have to add it to the SingletonScriptableObject  loader script

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Player/PlayerNetworkActions.cs]: ScriptableObjectSingletons in there

Comment (corp-0) [reply UnityProject/Assets/Scripts/Player/PlayerNetworkActions.cs]: Yes, I can do that. Would it work if I add a scriptable object that contains a list of occupations? So we don’t have to add all jobs to the Singleton list

Comment (corp-0) [reply UnityProject/Assets/Scripts/Player/PlayerNetworkActions.cs]: done

**Comment by corp-0** [comment]
I'm taking the Occupation reference from a singleton list now. Also fixed a problem I didn't notice before, when trying to respawn if you have no spawn point mapped you would just bug out of existence. It defaults to arrivals now but still can return null if arrivals hasn't been mapped and logs the error.

**Comment by Bod9001** [comment]
all good

**Pull request merged**